KMB/make_KMB_info.py

```diff
@@ -0,0 +1,270 @@
+        template_data['ID'] = item.ID
+        template_data['bildbeteckning'] = item.bildbeteckning
+        template_data['notes'] = ''
+        txt = helpers.output_block_template(template_name, template_data, 0)
+
+        # append object location if appropriate
+        if item.latitude and item.longitude:
+            txt += '\n{{Object location dec|%s|%s}}' % (
+                item.latitude, item.longitude)
+
+        return txt
+
+    def get_original_filename(self, item):
+        """
+        Return the url from which the image is fetched.
+
+        :param item: the KMBItem
+        :return: str
+        """
+        return item.source
+
+    # @todo: load mapping list instead of substing template - T164567
+    def get_photographer(self, item):
+        """
+        Return a correctly formated photographer value
+
+        :param item: the KMBItem
+        :return: str
+        """
+        template = 'User:Lokal_Profil/nycklar/creators'
+        return '{{safesubst:%s|%s|t}}' % (template, item.byline)
+
+    # @todo:
+    # * use fmis + bbr for (cached) commonscat searches on wikidata. And if
+    #   found then skip various other. - T164572
+    # * move some of this to KMBItem and set cats at the same time as the
+    #   text is processed.
+    # * Implement taggs from mapping - T164567
+    # * Add parish/municipality categorisation when needed - T164576
+    def generate_content_cats(self, item):
+        """
+        Extract any mapped keyword categories or depicted categories.
+
+        :param item: the KMBItem to analyse
+        :return: list of categories (without "Category:" prefix)
+        """
+        cats = []
+
+        # depicted
+        if item.fmis:
+            cats.append('Archaeological monuments in %s' % item.landskap)
+            cats.append('Archaeological monuments in %s County' % item.lan)
+        if item.bbr:
+            cats.append('Protected buildings in Sweden')
+
+        # must be better to do this via safesubst
+        for tagg in item.tagg:
+            # @todo: '{{safesubst:User:Lokal_Profil/nycklar/cats|%s|%s|%s}}' % (tagg, item.land.upper(), item.landskap)
+            pass
+
+        cats = list(set(cats))  # remove any duplicates
+        return cats
+
+    # @todo: Implement creator from mapping - T164567
+    def generate_meta_cats(self, item, content_cats):
+        """
+        Produce maintanance categories related to a media file.
+
+        :param item: the metadata for the media file in question
+        :param content_cats: any content categories for the file
+        :return: list of categories (without "Category:" prefix)
+        """
+        pass
+        cats = []
+
+        # base cats
+        # "Images from the Swedish National Heritage Board" already added by
+        # Kulturmiljöbild-image template
+        cats.append(self.batch_cat)
+
+        # problem cats
+        if not content_cats:
+            cats.append(self.make_maintanance_cat('improve categories'))
```

Comment:
Done.

This also triggered lokal-profil/BatchUploadTools#22 and https://github.com/lokal-profil/upload_batches/commit/91b57addc8aeaaeef4bfb8ec197e0a5e9af2248c